k8s/src/main/scala/io/buoyant/k8s/Watchable.scala

```diff
@@ -50,6 +38,37 @@ private[k8s] abstract class Watchable[O <: KubeObject: TypeReference, W <: Watch
     RetryBudget.Infinite
   )
 
+  def get(
+    labelSelector: Option[String] = None,
+    fieldSelector: Option[String] = None,
+    resourceVersion: Option[String] = None,
+    retryIndefinitely: Boolean = false,
+    watch: Boolean = false
+  ): Future[G] = {
+    val req = Api.mkreq(
+      http.Method.Get,
+      if (watch) watchPath else path,
+      None,
+      "labelSelector" -> labelSelector,
```

Comment:
tioli:

i tend to constify strings -- with, somewhere on an object:
```scala
   private[pkgname] val LabelSelector = "labelSelector"
```

only because typos become compile errors.

we'd want to use these static keys throughout